src/cursor/cur_file.c

```diff
@@ -469,15 +469,20 @@ __curfile_close(WT_CURSOR *cursor)
 
 	cbt = (WT_CURSOR_BTREE *)cursor;
 	CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, close, cbt->btree);
-	released = false;
+err:
 
 	/*
-	 * If releasing the cursor fails in any way, it will be left
-	 * in a state that allows it to be normally closed.
+	 * If releasing the cursor fails in any way, it will be left in a state
+	 * that allows it to be normally closed.
 	 */
 	WT_TRET(__wt_cursor_cache_release(session, cursor, &released));
```

Comment:
@keithbostic, I don't think we should be caching a cursor in an error path.  I pushed an alternative branch for your consideration.